docs/classic-ui/images.md

```diff
@@ -0,0 +1,114 @@
+---
+html_meta:
+  "description": ""
+  "property=og:description": ""
+  "property=og:title": ""
+  "keywords": ""
+---
+
+(classic-ui-images-label)=
+
+# Image resolving and scaling
+
+Image scaling is done in plone.namedfile.scaling and most interesting the ImageScaling view.
+
+## Image tag
+
+To get a HTML tag for an scaled image, you can use the ImageScaling view as follow:
+
+```python
+from plone import api
+
+scale_util = api.content.get_view('images', context, request)
+tag = scale_util.tag('leadimage', scale='mini')
+```
+
+or to get a specific size:
+
+```python
+from plone import api
+
+scale_util = api.content.get_view('images', context, request)
+tag = scale_util.tag('leadimage', width="600", height="200")
+```
+
+The first argument is the field name, followed by one of the following arguments:
+
+	fieldname=None,
+        scale=None,
+        height=None,
+        width=None,
+        direction="thumbnail",
+
+
+## Image scaling without tag creation
+
+To just get the scaling infos, without creating a HTML tag, you can use the ImageScaling view as follow:
+
+```python
+from plone import api
+
+scale_util = api.content.get_view('images', context, request)
+image_scale = scale_util.scale('leadimage', scale='mini')
+print(image_scale.url)  # http://localhost:8080/Plone/enl/@@images/3d182f34-8773-4f20-a79d-8774c3151b7e.jpeg
+print(image_scale.width)  # 200
+print(image_scale.height)  # 110
+```
+
+the most important properties:
+
+- data
+- fieldname
+- height
+- mimetype
+- srcset
+- srcset_attribute
+- tag
+- uid
+- url
+- width
+
+You can get a tag from this point too:
+```python
+>>> print(image_scale.tag)
+
+<img src="http://localhost:8080/Plone/news/newsitem1/@@images/9f676d46-0cb3-4512-a831-a5db4079bdfa.jpeg" alt="News Item 1!" title="News Item 1" height="21" width="32" srcset="http://localhost:8080/Plone/news/newsitem1/@@images/4a68513c-cffd-4de0-8a35-80627945b80f.jpeg 2x, http://localhost:8080/Plone/news/newsitem1/@@images/c32929c6-cb89-4ce7-846f-38adf29c09a4.jpeg 3x" />
+```
+
+or to get a specific size:
+
+```python
+from plone import api
+
+scale_util = api.content.get_view('images', context, request)
+tag = scale_util.scale('leadimage', width="600", height="200")
+```
+
+The first argument is the field name, followed by one of the following arguments:
+
+	fieldname=None,
+        scale=None,
+        height=None,
+        width=None,
+        direction="thumbnail",
```

Comment:
If `direction` is deprecated, should we include it? Should we replace it with `mode`?

Reply:
Also this is duplicated from earlier.

Reply:
Docs should use `mode`.

Reply:
right now, only plone.scaling implements mode, plone.namedfile which is the main package we are dealing with here, does not yet ;). So we have to use the old direction parameter until, this is refactured.

Reply:
OK. That makes sense by itself. However the section "Scaling `direction` deprecated in favor of `mode`" creates confusion. It appears to override everything written up to that point.

We could add [`version-changed`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionchanged) and [`deprecated`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated) directives to make it clear.

Reply:
I think we either refacture it right away or we don't mention the chnage for now, as the api of plone.namedfile has not changed yet. The rest is internal for now.

Reply:
Let's put this section in an `attention` admonition. That will both save the content for when y'all get around to refactoring it and notify users that this change is coming. I think we can also tune the language to make it clear that it is not ready yet.

Reply:
I pushed a commit https://github.com/plone/documentation/pull/1192/commits/58481e400f28d62b1ad48bdf33572b2710c6f81c that should address this.